diff --git a/src/ko/blacktoon/build.gradle.kts b/src/ko/blacktoon/build.gradle.kts
index dc49114029..76acc9e5e0 100644
--- a/src/ko/blacktoon/build.gradle.kts
+++ b/src/ko/blacktoon/build.gradle.kts
@@ -4,14 +4,14 @@ plugins {
 
 keiyoushi {
     name = "BlackToon"
-    versionCode = 10
+    versionCode = 11
     contentWarning = ContentWarning.NSFW
     libVersion = "1.4"
 
     source {
         name = "블랙툰"
         lang = "ko"
-        baseUrl = "https://blacktoon415.com"
+        baseUrl = "https://blacktoon417.com"
         skipCodeGen = true
     }
 }
diff --git a/src/ko/blacktoon/src/eu/kanade/tachiyomi/extension/ko/blacktoon/BlackToon.kt b/src/ko/blacktoon/src/eu/kanade/tachiyomi/extension/ko/blacktoon/BlackToon.kt
index 4f2a70bfaa..9ce026476c 100644
--- a/src/ko/blacktoon/src/eu/kanade/tachiyomi/extension/ko/blacktoon/BlackToon.kt
+++ b/src/ko/blacktoon/src/eu/kanade/tachiyomi/extension/ko/blacktoon/BlackToon.kt
@@ -341,7 +341,7 @@ abstract class BlackToon :
 
     companion object {
         private const val SOURCE_ID = 7_080_800_841_003_944_426L
-        private const val DEFAULT_DOMAIN_NUMBER = "415"
+        private const val DEFAULT_DOMAIN_NUMBER = "417"
         private const val DOMAIN_NUMBER_PREF = "domain_number"
         private const val USER_AGENT = "okhttp/4.12.0"
         private val DOMAIN_NUMBER_REGEX = Regex("""\d+""")
